### ISSUE TITLE
Add dark mode

### ISSUE THREAD

**Issue by aos** [aos/temple#16]
Have a dark mode version, and an ability to switch to it.

**Comment by oshikiri** [comment]
Hi, @aos.
Thanks for your cool Hugo theme.

I am trying to add a dark mode to this theme.
At first, I refactored CSS definitions. If the PR #32 is ok, I will add the dark mode definitions like below.

```css
@media (prefers-color-scheme: light), (prefers-color-scheme: no-preference) {
  :root {
    --color: #484848;
    --color-footnote: #575757;
    --color-footer-content: #bbb;
    --background-color: rgb(252, 252, 252);
    --background-color-code: rgb(240, 240, 240);
    --background-color-toggle-content: rgb(249, 249, 249);
    --link-color: #c05b4d;
    --link-color-hover: #a5473a;
    --list-link-color: #369;
    --border-navbar-and-footer: #e3e3e3;
  }
}

@media (prefers-color-scheme: dark) {
  :root {
    --color: rgb(200, 200, 200);
    --color-footnote: rgb(150, 150, 150);
    --color-footer-content: #bbb;
    --background-color: rgb(5, 5, 5);
    --background-color-code: rgb(30, 30, 30);
    --background-color-toggle-content: rgb(10, 10, 10);
    --link-color: rgb(0, 210, 238);
    --link-color-hover: rgb(0, 141, 160);
    --list-link-color: rgb(72, 134, 197);
    --border-navbar-and-footer: #e3e3e3;
  }
}
```

What do you think?

**Comment by aos** [comment]
This is a great idea! I like what you are doing here 👍 I replied to your PR. Let me know your thoughts.

**Comment by oshikiri** [comment]
Thanks :smile: 

I randomly chose the colors of dark mode in the above snippet.
If you want to tweak the colors, please feel free to edit and push the snippet.

How about that?

**Comment by aos** [comment]
Sounds good -- lemme take a look at them sometime tomorrow and report back

**Comment by aos** [comment]
Hey @oshikiri -- I didn't get a chance to look at it yesterday. Do you want to throw a PR up and we can work from there? One thing to keep in mind is syntax highlighting: it is generated at build time with Hugo ([using chroma](https://gohugo.io/content-management/syntax-highlighting/)). I'm not exactly sure how we can make it dynamic between light/dark mode. But I'm OK with keeping the `pre`/`code`, any code blocks and any other code as is without going into dark mode.

**Comment by oshikiri** [comment]
>Do you want to throw a PR up and we can work from there?

OK. I created the PR https://github.com/aos/temple/pull/33.

**Comment by aos** [comment]
Implemented in #35
